fuzz_test.go

```diff
@@ -504,3 +504,29 @@ func TestFuzz_SkipPattern(t *testing.T) {
 		return 5, true
 	})
 }
+
+func TestNewFromGoFuzz(t *testing.T) {
```

Comment:
This is actually a super important test-- we need to promise callers that `data` they find by adversarial fuzzing continues to always work. This test checks that, and will catch e.g. if go changes something about the random number generator (in which case, we'd have to fork and/or depend on an older version to keep all the output exact).